frappe/public/js/frappe/list/list_view.js

```diff
@@ -818,16 +818,41 @@ frappe.views.ListView = class ListView extends frappe.views.BaseList {
 		let html = "";
 
 		let settings_button = null;
-		if (this.settings.button && this.settings.button.show(doc)) {
-			settings_button = `
-				<span class="list-actions">
-					<button class="btn btn-action btn-default btn-xs"
-						data-name="${doc.name}" data-idx="${doc._idx}"
-						title="${this.settings.button.get_description(doc)}">
-						${this.settings.button.get_label(doc)}
-					</button>
-				</span>
-			`;
+
+		// check if the button property is an array or object
+		if (Array.isArray(this.settings.button)) {
+			settings_button = "";
+			for (const button of this.settings.button) {
+				if (!button.name) {
+					frappe.throw(__("Button needs a unique 'name' when using multiple buttons."));
+				}
+
+				if (button && button.show(doc)) {
+					settings_button += `
+                        <span class="list-actions">
+                            <button class="btn btn-action btn-default btn-xs"
+                                data-name="${doc.name}" data-idx="${doc._idx}" data-button-name="${
+						button.name
+					}"
+                                title="${button.get_description(doc)}">
+                                ${button.get_label(doc)}
+                            </button>
+                        </span>
+                    `;
```

Comment:
This code is duplicated. Better to put this in a separate function, even if internal to this function.

Reply:
Even simpler option: if not array, create an array with single button in it so rest of the code is same and you don't need 2 different branches.

Reply:
Thanks. I've updated that. Also, how can I make sure each button has a unique name? I think it would save a lot of debugging headache if we could show a warning that there are non-unique names.